src/google/protobuf/compiler/php/php_generator.cc

```diff
@@ -139,31 +139,44 @@ std::string ClassNamePrefix(const string& classname,
   return "";
 }
 
+template <typename DescriptorType>
+std::string PhpNamespace(const DescriptorType* desc, bool is_descriptor) {
+  if (desc->file()->options().has_php_namespace()) {
+    const string& php_namespace = desc->file()->options().php_namespace();
+    if (php_namespace != "") {
+      return php_namespace;
+    } else {
+      return "";
+    }
+  }
+
+  if (desc->file()->package() == "") {
+    return "";
+  } else {
+    return PhpName(desc->file()->package(), is_descriptor);
+  }
+}
 
 template <typename DescriptorType>
-std::string FullClassName(const DescriptorType* desc, bool is_descriptor) {
+std::string PhpClassName(const DescriptorType* desc, bool is_descriptor) {
   string classname = desc->name();
   const Descriptor* containing = desc->containing_type();
   while (containing != NULL) {
     classname = containing->name() + '_' + classname;
     containing = containing->containing_type();
   }
-  classname = ClassNamePrefix(classname, desc) + classname;
+  return ClassNamePrefix(classname, desc) + classname;
+}
 
-  if (desc->file()->options().has_php_namespace()) {
-    const string& php_namespace = desc->file()->options().php_namespace();
-    if (php_namespace != "") {
-      return php_namespace + '\\' + classname;
-    } else {
-      return classname;
-    }
-  }
+template <typename DescriptorType>
+std::string FullClassName(const DescriptorType* desc, bool is_descriptor) {
```

Comment:
This refactoring of FullClassName seems like it is from https://github.com/google/protobuf/pull/3338 but it isn't used in this PR